OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -99,65 +97,44 @@ bool CanRearm()
 			return rearmable != null && rearmable.RearmableAmmoPools.Any(p => !p.FullAmmo());
 		}
 
-		public string VoicePhraseForOrder(Actor self, Order order)
+		string IOrderVoice.VoicePhraseForOrder(Actor self, Order order)
 		{
 			return order.OrderString == "Repair" && (CanRepair() || CanRearm()) ? Info.Voice : null;
 		}
 
 		void IResolveOrder.ResolveOrder(Actor self, Order order)
 		{
-			if (order.OrderString == "Repair")
-			{
-				// Repair orders are only valid for own/allied actors,
-				// which are guaranteed to never be frozen.
-				if (order.Target.Type != TargetType.Actor)
-					return;
+			if (order.OrderString != "Repair")
+				return;
 
-				// Aircraft handle Repair orders directly in the Aircraft trait
-				if (self.Info.HasTraitInfo<AircraftInfo>())
-					return;
+			// Repair orders are only valid for own/allied actors,
+			// which are guaranteed to never be frozen.
+			if (order.Target.Type != TargetType.Actor)
+				return;
 
-				if (!CanRepairAt(order.Target.Actor) || (!CanRepair() && !CanRearm()))
-					return;
+			// Aircraft handle Repair orders directly in the Aircraft trait
+			if (self.Info.HasTraitInfo<AircraftInfo>())
```

Comment:
You could cache a `bool isAircraft` from `INotifyCreated.Created` above.

...or maybe remove this Aircraft special case completely, like we discussed the other day? I assume that didn't go so well...

Reply:
> I assume that didn't go so well...

Nope, another can of worms. Going by experience, this looks like something that could easily take 4-8 weeks to get right (delays in reviewing and updating calculated in).